### DISCUSSION TITLE
Delete permission per DAG instead of all DAGs

### DISCUSSION THREAD

**Discussion by thoralf-gutierrez** [apache/airflow#19634] [General]
Hello everybody!

We've been using Airflow successfully with different teams on one Airflow cluster, with a custom security manager that assigns roles to users based on their LDAP groups. Each role is then able to see/edit certain DAGs if the DAG owner correspond to the role/LDAP group. We do this with one resource per DAG (e.g. "DAG:example_dag") and our security manager sync role permissions base on dag ownership every so often.

We would like to also allow teams to delete their own DAGs, but unfortunately permission.ACTION_CAN_DELETE only applies to the "DAGs" resource, which gives a user the permission to delete _any_ DAG. We haven't found a way to add the permission.ACTION_CAN_DELETE to individual "DAG:*" resources.

Is there a plan to add delete permissions for individual dag resources? How do other people manage this?

Thanks,
Thor

**Comment by potiuk** [comment]
This is going to be likely implemented in the future, there is only dag-level access (per dag). There are no "group" access levels.

There is work around multi-tenancy and "resource level" access is one of the features we are going to discuss soon - see  https://docs.google.com/document/d/17kgfLO2fpNC62YuCxo1l0yRipt48YE6g1d1SGUok2I8/edit#heading=h.d14mqct3autb
and https://apache-airflow.slack.com/archives/C02M551UDA4

**Reply by thoralf-gutierrez** [reply]
To be clear, I'm not interested in "group" access levels. I manage that by creating dag-level permissions for each dag, and syncing roles to make sure they each role has access to all the dags they need to. What I am saying is that the _delete_ dag-level permission seems to be missing, and I wonder why that's the case and if it's on the roadmap.

**Reply by potiuk** [reply]
Ah. This I am not sure but maybe @jhtimmins  knows ?

**Reply by thoralf-gutierrez** [reply]
@jhtimmins do you know how we could do this, or if it is on the roadmap ?
